CIP-????/README.md

```diff
@@ -0,0 +1,530 @@
+
+- Have the transaction piece reference a particular input that can be spent by the user.
+- Have the transaction piece spend a particular input that can also be directly spent by the user.
+
+With either of the above choices, spending the input in question will invalidate all transaction
+pieces with this input since the UTxO would no longer exist. Both types of kill-switches can even be
+used within the same transaction piece. Using both would give users the flexibility of either
+canceling a particular limit order or a batch of limit orders (eg, all limit orders for a particular
+trading pair).
+
+The aggregator can take all (still valid) transaction pieces from users and batch them together,
+possibly with on-chain liquidity sources, to submit them in a single transaction. Since the
+transaction pieces did not need to specify a counter-party, these trades can be matched against
+**any** counter-party, no matter its form. It is even trivially possible to match a single large
+order against several smaller orders.
+
+With this approach, all high frequency activity can happen off-chain while settling the actual
+trades on-chain in the most efficient way possible.
+
+This approach can even be combined with Babel Fees to enable day-traders to pay the transaction fee
+in a native asset specified by the aggregator.
+
+## Specification
+
+Transaction pieces **must** be included all-or-nothing, and a transaction would become
+**exclusively** a list of signed pieces (ie, an output cannot be specified outside of a transaction
+piece). The current one-party style transaction can still be represented by a transaction with only
+one piece where that piece is a fully balanced transaction body.
+
+### CDDL
+
+#### Transaction
+
+Below is the current cddl representation of a transaction:
+
+```cddl
+transaction =
+  [ transaction_body
+  , transaction_witness_set
+  , bool
+  , auxiliary_data / null
+  ]
+```
+
+It contains a single transaction body and a single witness set. This will be changed to contain a
+list of transaction bodies, each paired with their own witness set.
+
+The transaction witness set is currently:
+
+```cddl
+transaction_witness_set =
+  { ? 0: [* vkeywitness ]
+  , ? 1: [* native_script ]
+  , ? 2: [* bootstrap_witness ]
+  , ? 3: [* plutus_v1_script ]
+  , ? 4: [* plutus_data ]
+  , ? 5: [* redeemer ]
+  , ? 6: [* plutus_v2_script ] ; New
+  }
+```
+
+Smart contracts will still be executed against the full transaction (this will be discussed in a
+later section), but all of the required information for an execution (eg, script, datum, and
+redeemer) can be found within the corresponding transaction piece. This enables the witness set to
+remain unchanged.
+
+The new transaction would then be:
+
+```cddl
+transaction_piece =
+  { 0: transaction_body
+  , 1: transaction_witness_set ; The witness set for this transaction body.
+  }
+
+transaction =
+  [ transaction_pieces    : [* transaction_piece]
+  , bool
+  , auxiliary_data / null
```

Comment:
I thought I mentioned that, but I guess I accidentally deleted that part while editing. Each piece *can* specify it, but if multiple do, they must all agree. I wasn't anticipating users to use pieces for storing metadata on-chain. I thought they would build a personal transaction for that.

Reply:
So it is the "must agree" thing sprinkled throughout the CIP that I really have a problem with. Because of those agreements would have to be enforced by the ledger rules. Which increases complexity of already very complex rules.